util/TLSProxy/Message.pm

```diff
@@ -198,14 +198,14 @@ sub get_messages
         print "  [".$record->decrypt_data."]\n";
     } elsif ($record->content_type == TLSProxy::Record::RT_ALERT) {
         my ($alertlev, $alertdesc) = unpack('CC', $record->decrypt_data);
-        #All alerts end the test
-        $end = 1;
         #A CloseNotify from the client indicates we have finished successfully
         #(we assume)
-        if (!$server && $alertlev == AL_LEVEL_WARN
+        if (!$end && !$server && $alertlev == AL_LEVEL_WARN
             && $alertdesc == AL_DESC_CLOSE_NOTIFY) {
             $success = 1;
         }
+        #All alerts end the test
+        $end = 1;
```

Comment:
It doesn't appear that the server ever sends close_notify. Or, at least, if it does, the proxy never sees it.

(Interestingly, where I get one alert or two alerts from the client seems to be somewhat flaky... I'm guessing it depends on timing somehow. Maybe if the two alerts got into one read?)